tests/integration/Settings_API/SettingTest.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+use SkyVerge\WooCommerce\PluginFramework\v5_6_1\Settings_API\Setting;
+
+class SettingTest extends \Codeception\TestCase\WPTestCase {
+
+
+	/** @var \IntegrationTester */
+	protected $tester;
+
+
+	protected function _before() {
+
+	}
+
+
+	protected function _after() {
+
+	}
+
+
+	/** Tests *********************************************************************************************************/
+
+
+	/**
+	 * @see Setting::validate_value()
+	 *
+	 * @param mixed $value value to pass to method
+	 * @param string $type setting type
+	 * @param bool $type whether the value should be considered valid or not
+	 *
+	 * @dataProvider provider_validate_value
+	 * */
+	public function test_validate_value( $value, $type, $expected ) {
+
+		$setting = new Setting();
+		$setting->set_type( $type );
+
+		$this->assertSame( $expected, $setting->validate_value( $value ) );
+	}
+
+
+	/**
+	 * Provider for test_validate_value()
+	 *
+	 * @return array
+	 */
+	public function provider_validate_value() {
+
+		require_once( 'woocommerce/Settings_API/Setting.php' );
+
+		return [
+			[ 'example', Setting::TYPE_STRING, true ],
+			[ 3.1415926, Setting::TYPE_STRING, false ],
+
+			[ 'https://skyverge.com/', Setting::TYPE_URL, true ],
+			[ 'file:///tmp/', Setting::TYPE_URL, false ],
+			[ 'example', Setting::TYPE_URL, false ],
+
+			[ 'test@example.com', Setting::TYPE_EMAIL, true ],
+			[ 'not-an-email.com', Setting::TYPE_EMAIL, false ],
+			[ '', Setting::TYPE_EMAIL, false ],
+
+			[ 1729, Setting::TYPE_INTEGER, true ],
+			[ 'hi', Setting::TYPE_INTEGER, false ],
```

Comment:
Do we want to accept numeric strings as integers? As the code is now, I believe `'1729'` would be considered valid. Just want to make sure that is what we want.

Reply:
I think we don't want to accept numeric strings. So it should behave as if `declare(strict_types=1)` was used. We already force booleans to be bool, so it makes sense to do the same here.

The default implementation of `get_value_from_database()` converts values for boolean settings only, so to be strict we may need to start converting ints and floats there as well. 

I can make those updates too, if that sounds right.

Reply:
I am good with both updates. @ChaseWiseman ?

Reply:
Sounds right to me